pkg/kubelet/custommetrics/custom_metrics.go

```diff
@@ -36,7 +36,7 @@ const (
 func GetCAdvisorCustomMetricsDefinitionPath(container *v1.Container) (*string, error) {
 	// Assumes that the container has Custom Metrics enabled if it has "/etc/custom-metrics" directory
 	// mounted as a volume. Custom Metrics definition is expected to be in "definition.json".
-	if container.VolumeMounts != nil {
+	if container != nil {
```

Comment:
```suggestion
	if container != nil && container.VolumeMounts != nil {
```

Reply:
I think that calling `range` on nil is the same as calling `range` on an empty list - so I'm not sure checking `container.VolumeMounts != nil` is necessary.

Can't hurt to do though.

Reply:
@feiskyer I think slice nil check is not necessary before range. see the example: https://play.golang.org/p/uzFAZgb8Azg. What worried me is the nil pointer dereference, it will cause runtime panic, compiler can do nothing in this situation. @mattjmcnaughton as the demo shows, it's not hurt to range nil slice :)